sdk/core/core-amqp/src/errors.ts

```diff
@@ -547,47 +610,62 @@ function isBrowserWebsocketError(err: any): boolean {
   return result;
 }
 
+const rheaPromiseErrors = [
+  // OperationTimeoutError occurs when the service fails to respond within a given timeframe.
+  "OperationTimeoutError",
+
+  // InsufficientCreditError occurs when the number of credits available on Rhea link is insufficient.
+  "InsufficientCreditError",
+
+  // Defines the error that occurs when the Sender fails to send a message.
+  "SendOperationFailedError"
+];
+
 /**
- * Translates the AQMP error received at the protocol layer or a generic Error into a MessagingError.
+ * This utility checks if an error is a retryable error from rhea-promise.
+ * @param err
+ */
+function isRheaPromiseError(err: Error): boolean {
+  return rheaPromiseErrors.indexOf(err.name) !== -1;
+}
+
+/**
+ * Translates the AQMP error received at the protocol layer or a SystemError into a MessagingError.
+ * All other errors are returned unaltered.
  *
  * @param {AmqpError} err The amqp error that was received.
  * @returns {MessagingError} MessagingError object.
  */
-export function translate(err: AmqpError | Error): MessagingError {
-  if ((err as MessagingError).translated) {
+export function translate(err: AmqpError | Error): MessagingError | Error {
+  if ((err as MessagingError).name === "MessagingError") {
```

Comment:
If we move this check to after `if (isAmqpError)` then we can skip the casting to `MessagingError` and simply

```javascript
if (err.name === "MessagingError") {
    // already translated.
    return err;
  }
```

Reply:
Done